### PR TITLE
devguide: add page on installing suricata from git

### PR DESCRIPTION
**Pull request by mrbazzan**
Continuation from #8059

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [#5585](https://redmine.openinfosecfoundation.org/issues/5585)

Describe changes:
- Changed the order of `installation-from-git` to have a better semantic order
- Reworded the `Post installation` sub-section
- Fixed the reviews from #8059 

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Review by jufajardini — APPROVED** [review]
Feedback given by Jeff was incorporated. I think that this bit is a bit confusing, but understandable, especially for someone who has followed the whole guide, with the rewording...

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+.. note:: If you would like to build ``suricata-update`` from source, enter:
+
+  .. code-block:: bash
+
+    sudo apt install -y python3 python3-distutils python3-yaml
+    ./scripts/bundle.sh suricata-update
+    cd suricata-update
+    python3 setup.py build
+    sudo python3 setup.py install
+    sudo suricata-update
```
Comment (jasonish) [inline doc/userguide/devguide/codebase/installation-from-git.rst:116]: I'd get rid of this.  Installing Suricata and Suricata-Update separately does result in an installation that doesn't match our documentation, training, etc.

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  ./autogen.sh
+
+
+To configure, please enter:
+
+.. code-block:: bash
+
+  ./configure
+
+
```
Comment (jasonish) [inline doc/userguide/devguide/codebase/installation-from-git.rst:88]: Extra line.

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  ./scripts/bundle.sh libhtp
+
+Followed by:
+
+.. code-block:: bash
+
+  ./autogen.sh
+
+
```
Comment (jasonish) [inline doc/userguide/devguide/codebase/installation-from-git.rst:81]: Extra line.

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  git clone https://github.com/OISF/suricata.git
+  cd suricata
+
+Libhtp is not bundled. Get libhtp by doing:
+
+.. code-block:: bash
+
+  ./scripts/bundle.sh libhtp
+
```
Comment (jasonish) [inline doc/userguide/devguide/codebase/installation-from-git.rst:74]: I think we should go ahead and just bundle both suricata-update and libhtp here.  This gets you an installation that is inline with our documentation and usage guides.  As well as how are QA is setup.  And is required for `make install-full` and such as documented below.

**Comment by mrbazzan** [comment]
Continued on #8128

**Pull request closed without merging**